habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomMonorepoGroupMigration.java

```diff
@@ -0,0 +1,147 @@
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        replacements.clear();
+        hasExistingMonoRepoDependenciesGroup = false;
+
+        boolean shouldExecute = false;
+        try (FileConfig tomlFileConfig = FileConfig.of(file)) {
+            tomlFileConfig.load();
+
+            Optional<Config> toolPoetryDependencies = tomlFileConfig.getOptional(TomlUtils.TOOL_POETRY_DEPENDENCIES);
+            if (toolPoetryDependencies.isPresent()) {
+                Config foundDependencies = toolPoetryDependencies.get();
+                Map<String, Object> dependencyMap = foundDependencies.valueMap();
+
+                for (Map.Entry<String, Object> dependency : dependencyMap.entrySet()) {
+                    String packageName = dependency.getKey();
+                    Object packageRhs = dependency.getValue();
+                    if (TomlUtils.representsLocalDevelopmentVersion(packageRhs)) {
+                        String packageRshAsString = TomlUtils.convertCommentedConfigToToml((CommentedConfig) packageRhs);
+                        logger.info("Found local dependency not within monorepo group! ({} = {})", packageName, packageRshAsString);
+                        TomlReplacementTuple replacementTuple = new TomlReplacementTuple(packageName, packageRshAsString, "");
+                        replacements.put(packageName, replacementTuple);
+                        shouldExecute = true;
+                    }
+                }
+            }
+
+            Optional<Config> toolPoetryMonorepoDependencies = tomlFileConfig.getOptional(TomlUtils.TOOL_POETRY_GROUP_MONOREPO_DEPENDENCIES);
+            if (toolPoetryMonorepoDependencies.isPresent()) {
+                hasExistingMonoRepoDependenciesGroup = true;
+            }
+        }
+
+        return shouldExecute;
+    }
+
+    @Override
+    protected boolean performMigration(File pyProjectTomlFile) {
+        String fileContent = StringUtils.EMPTY;
+        try (BufferedReader reader = new BufferedReader(new FileReader(pyProjectTomlFile))) {
+            String line = reader.readLine();
+            boolean injectAfterNextEmptyLine = false;
+
+            while (line != null) {
+                boolean addLine = true;
+                boolean isEmptyLine = line.isBlank();
+
+                if (line.contains(StringUtils.SPACE) || line.contains(TomlUtils.EQUALS)) {
+                    String key = line.substring(0, line.indexOf(StringUtils.SPACE));
+
+                    if (key == null) {
+                        key = line.substring(0, line.indexOf(TomlUtils.EQUALS));
+                    }
+
+                    if (key != null) {
+                        key = key.strip();
+
+                        TomlReplacementTuple matchedTuple = replacements.get(key);
+                        if (matchedTuple != null) {
+                            // skip this line, we will add it back to [tool.poetry.group.monorepo.dependencies] later
```

Comment:
A: Do we have an open issue for this yet, just so we keep track?

Reply:
No need - it does that later in the code block.  It's just a note about how the processing will work.

Reply:
Ignore, just got to the part where this is addressed